pybamm/spatial_methods/finite_volume.py

```diff
@@ -342,6 +342,71 @@ def definite_integral_vector(self, domain):
 
         return pybamm.Vector(vector)
 
+    def indefinite_integral(self, domain, symbol, discretised_symbol):
+        """Implementation of the indefinite integral operator. The
+        input discretised symbol must be defined on the internal mesh edges.
+        See :meth:`pybamm.BaseDiscretisation.indefinite_integral`
+        """
+
+        if not symbol.has_gradient_and_not_divergence():
```

Comment:
we've got a lot of these little functions in `pybamm.Symbol` now. Might be worth consolidating them, for example into:
```python
def has_node_types(self, list_of_types):
   # implementation here
``` 

this is an aside, not for this issue...

Reply:
Yeah that's a good idea, would be very helpfully in quite a few places in the discretisation and spatial methods